src/components/TodoList/TodoItem.tsx

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { useState, useEffect } from 'react';
+import { observer } from 'mobx-react';
+import classNames from 'classnames';
+import useStore from '../../hooks/useStore';
+import { ITodo } from '../../types/todoTypes';
+
+const TodoItem = ({ content, completed, id }: ITodo) => {
+  const [isChecked, setIsChecked] = useState<boolean>(completed);
+  const [renderFlag, setRenderFlag] = useState<boolean>(true);
+  const [isEditing, setIsEditing] = useState<boolean>(false);
+  const [editedTodo, setEditedTodo] = useState<string>(content);
+  const { todoStore } = useStore();
+
+  useEffect(() => {
+    if (todoStore.path === '/completed') {
+      completed ? setRenderFlag(true) : setRenderFlag(false);
+    } else if (todoStore.path === '/active') {
+      completed ? setRenderFlag(false) : setRenderFlag(true);
```

Comment:
```suggestion
    if (todoStore.path === '/completed') {
      isChecked ? setRenderFlag(true) : setRenderFlag(false);
    } else if (todoStore.path === '/active') {
      isChecked ? setRenderFlag(false) : setRenderFlag(true);
```
해당 부분에서 2가지 의견이 있습니다.

첫번째, `const [isChecked, setIsChecked] = useState<boolean>(completed);`로 `completed`를 isChecked로 사용하고  있는데 우선 isChecked로 state를 선언해주었음에도 completed를  사용하는건 여러가지 측면에서 안좋아보입니다.

또한, 이렇게 코드를 작성하면 체크를 해제했을때, item이 완료되지 않은거니까 사라져야하는데 사라지지 않습니다. (`active`도 마찬가지)
<img src="https://user-images.githubusercontent.com/22424891/140022841-9a7f6bb4-77e6-4bb2-a076-2157b51d8177.png" height="300px" />

두번째는, footer에서 href 속성을 사용해서, useEffect 코드가 항상 실행되는데, `useEffect(() => {`로 얻을 수 있는 이점이 없다고 생각합니다. (그냥 함수로도 충분..?) useEffect를 사용하신 이유가 따로 있을까요?

```js
      <a href={path} className={className}>
        {text}
      </a>
```